[virt] DPDK Checkup: Update user facing API (#62967)
openshift/openshift-docs: modules/virt-dpdk-config-map-parameters.adoc

|True

|`spec.param.trafficGenContainerDiskImage`
|The container disk image for the traffic generator. The default value is `quay.io/kiagnose/kubevirt-dpdk-checkup-traffic-gen:main`.
|False

|`spec.param.trafficGenTargetNodeName`
|The node on which the traffic generator VM is to be scheduled. The node should be configured to allow DPDK traffic.
|False

|`spec.param.trafficGenPacketsPerSecond`
|The number of packets per second, in kilo (k) or million(m). The default value is 8m.
|False

|`spec.param.vmUnderTestContainerDiskImage`
|The container disk image for the VM under test. The default value is `quay.io/kiagnose/kubevirt-dpdk-checkup-vm:main`.
|False

|`spec.param.vmUnderTestTargetNodeName`
|The node on which the VM under test is to be scheduled. The node should be configured to allow DPDK traffic.
|False

|`spec.param.testDuration`
|The duration, in minutes, for which the traffic generator runs. The default value is 5 minutes.
|False

|`spec.param.portBandwidthGbps`
|The maximum bandwidth of the SR-IOV NIC. The default value is 10Gbps.
|False

|`spec.param.verbose`
|When set to `true`, it increases the verbosity of the checkup log. The default value is `false`.
|False
|====

> .DPDK checkup config map output fields

Author: Done.

[cols="1,1", options="header"]
|====
|Parameter
|Description

|`status.succeeded`

> |Has the checkup succeeded

Author: Done.

|`status.failureReason`

> |Failure reason in case of a failure

Author: Done.

|`status.startTimestamp`

> |Checkup start timestamp

Author: Done.

|`status.completionTimestamp`

> |Checkup completion timestamp

Author: Done.

|`status.result.trafficGenSentPackets`

> |Number of packets sent from the traffic generator

Author: Done.

|`status.result.trafficGenOutputErrorPackets`

> |Indicates error sending packets from traffic generator

Reviewer: ```suggestion
|The number of error packets sent from the traffic generator.
```

Author: Done.

|`status.result.trafficGenInputErrorPackets`

> |Indicates error receiving packets to traffic generator

Reviewer: ```suggestion
|The number of error packets received by the traffic generator.
```

Author: Done.

|`status.result.trafficGenActualNodeName`

> |Node name on which the traffic generator VM was scheduled

Author: Done.

|`status.result.vmUnderTestActualNodeName`

> |Node name on which the VM under test was scheduled

Author: Done.

|`status.result.vmUnderTestReceivedPackets`

> |Number of packets received on the VM under test

Author: Done.

> |Number of packets packets received on the VM under test

Reviewer 2: "packets" duplicated

Author: Good catch!
Done.

|`status.result.vmUnderTestRxDroppedPackets`

> |Indicates ingress packets that were dropped from DPDK application

Reviewer: ```suggestion
|The ingress traffic packets that were dropped by the DPDK application.
```

Author: Done.

|`status.result.vmUnderTestTxDroppedPackets`

> |Indicates egress packets were dropped from the DPDK application

Reviewer: ```suggestion
|The egress traffic packets that were dropped from the DPDK application.
```

Author: Done.
